mw_basic_auth.go

```diff
@@ -84,30 +91,37 @@ func (k *BasicAuthKeyIsValid) ProcessRequest(w http.ResponseWriter, r *http.Requ
 		return k.requestForBasicAuth(w, "User not authorised")
 	}
 
-	// Ensure that the username and password match up
-	var passMatch bool
 	switch session.BasicAuthData.Hash {
 	case user.HashBCrypt:
-		err := bcrypt.CompareHashAndPassword([]byte(session.BasicAuthData.Password), []byte(authValues[1]))
-		if err == nil {
-			passMatch = true
-		}
-	case user.HashPlainText:
-		if session.BasicAuthData.Password == authValues[1] {
-			passMatch = true
-		}
-	}
+		cacheKey := session.BasicAuthData.Password + authValues[1]
+		// check if username/password combination in cache
+		val, isCached := k.cache.Load(cacheKey)
+		if isCached {
 
-	if !passMatch {
-		logEntry.Info("Attempted access with existing user but failed password check.")
+			log.Debug("cache hit: ", cacheKey)
```

Comment:
Should we print cashed keys? (even in debug mode)?

Reply:
This is for me to test locally that my cache is working as expected.